src/sasl-sha1.js

```diff
@@ -1,8 +1,7 @@
 import MD5 from './md5';
 import SASLMechanism from './sasl.js';
-import SHA1 from './sha1';
```

Comment:
The `sha1.js` file can now be removed right?

Reply:
I believe so? I didn't see if it was used anywhere else.

Reply:
We could also probably get rid of md5 if we factor out the client challenge generation to `scram.js` as you suggested above. This would probably require the client challenge be marked as async, but since we're making everything else async, it's probably fine. Honestly, it probably should be async anyway. Let me know.

Reply:
> We could also probably get rid of md5 if we factor out the client challenge generation to `scram.js` as you suggested above. This would probably require the client challenge be marked as async, but since we're making everything else async, it's probably fine. Honestly, it probably should be async anyway. Let me know.

On a second look, it does not require async. We can leave it as is for now.

Reply:
SHA1 is exported from Strophe and pulled in as an implicit dependency in converse, for caps. I believe this is the only place where this happens. MD5 can be safely removed, though.

Reply:
> SHA1 is exported from Strophe and pulled in as an implicit dependency in converse, for caps.

Ah, I forgot about that. I've removed it now (took me 3 commits :sweat_smile: ).

I think we can remove both SHA and MD5 from Strophe and make the next release a major version bump.

Reply:
I see you've removed both already.

Reply:
> > SHA1 is exported from Strophe and pulled in as an implicit dependency in converse, for caps.
> 
> Ah, I forgot about that. I've removed it now (took me 3 commits sweat_smile ).
> 
> I think we can remove both SHA and MD5 from Strophe and make the next release a major version bump.

Yeah, I feel you there. Working with git is not my forte. I was working on this as well but ran into the failing tests issue, so thanks.